File: Dockerfile.py
FROM python:3.11-slim

WORKDIR /app

# Установка системных зависимостей
RUN apt-get update && apt-get install -y --no-install-recommends \
    gcc \
    g++ \
    && rm -rf /var/lib/apt/lists/*

# Копируем requirements и устанавливаем зависимости
COPY cmd/scoring-service/requirements.txt .
RUN pip3 install --no-cache-dir -r requirements.txt
RUN pip3 install grpcio grpcio-tools

# Устанавливаем spacy и модель для русского языка
RUN pip3 install spacy
RUN python3 -m spacy download ru_core_news_sm

# Копируем proto файлы
COPY proto /app/proto

# Генерируем Python код из proto
RUN python3 -m grpc_tools.protoc -I /app/proto --python_out=/app --grpc_python_out=/app /app/proto/nlp.proto

# Копируем исходный код сервиса
COPY cmd/scoring-service/main.py /app/

CMD ["python3", "main.py"]

File: Dockerfile.go
FROM golang:1.25-alpine AS builder

WORKDIR /app
COPY . .
RUN go mod download
RUN go build -o bin/api-gateway cmd/api-gateway/main.go

FROM alpine:latest
WORKDIR /app

RUN apk add --no-cache netcat-openbsd

COPY --from=builder /app/bin/api-gateway .
COPY wait-for.sh .
COPY .env .

CMD ["./wait-for.sh", "scoring-service", "50051", "./api-gateway"]